kuzzleio/kuzzle-build
# Mac OSX Setup.sh tests

## What does this PR do ?
This PR introduces the tests for the setup.sh script in a Mac OSX environment. The tests are run in a OSX machine in the cloud, so that Travis launches them via a ssh command. 

### How should this be manually tested?

  - clone this repository locally and checkout the `setupsh-test-mac` branch
  - set the following environment variables
    - `SETUPSH_TEST_DISTROS=osx`
    - `MAC_USER=xxx` (find it in our private accounts document)
    - `MAC_HOST=xxx` (find it in our private accounts document)
    - `TRAVIS_BRANCH=setupsh-test-mac`
    - `SETUPSH_SHOW_DEBUG=1` (useful if you want to see the debugging output of the tests)
    - Add the private key to your `~/.ssh` (find it in our private accounts document)
  - run `test/run.sh`

### Other changes

The tests in debian-like distributions were broken due to a fail while fetching the `curl` package. Adding `apt-get update` before each package install fixed the problem. Also, the setup.sh script used `which` to generate the system-v scripts, but `which` doesn't seem to be included by default in Fedora, so it has been replaced by `command -v`.

### Boyscout

There is no boyscouting in this PR.
